speechbrain/lobes/models/discrete/sq_codec.py

```diff
@@ -0,0 +1,1394 @@
+    out_channels : int
+        Number of output channels.
+    kernel_size : int
+        Size of the convolutional kernel.
+    stride : int, optional
+        Stride of the convolution (default is 1).
+    dilation : int, optional
+        Dilation factor for the convolution (default is 1).
+    groups : int, optional
+        Number of blocked connections from input channels to output channels (default is 1).
+    padding_mode : str, optional
+        Padding mode to use ('zeros', 'reflect', 'replicate', or 'circular') (default is 'zeros').
+    bias : bool, optional
+        If True, adds a learnable bias to the output (default is True).
+    padding : int, optional
+        Explicit padding value. If not provided, it will be computed automatically.
+    causal : bool, optional
+        If True, applies causal convolution where the output depends only on the past and current inputs (default is False).
+    w_init_gain : str, optional
+        Gain value used for Xavier initialization (e.g., 'relu', 'tanh', etc.). If provided, applies Xavier uniform initialization to the convolutional weights.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int = 1,
+        dilation: int = 1,
+        groups: int = 1,
+        padding_mode: str = "zeros",
+        bias: bool = True,
+        padding=None,
+        causal: bool = False,
+        w_init_gain=None,
+    ):
+        self.causal = causal
+        if padding is None:
+            if causal:
+                padding = 0
+                self.left_padding = dilation * (kernel_size - 1)
+            else:
+                padding = get_padding(kernel_size, dilation)
+        super(Conv1d, self).__init__(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=stride,
+            padding=padding,
+            dilation=dilation,
+            groups=groups,
+            padding_mode=padding_mode,
+            bias=bias,
+        )
+        if w_init_gain is not None:
+            torch.nn.init.xavier_uniform_(
+                self.weight, gain=torch.nn.init.calculate_gain(w_init_gain)
+            )
+
+    def forward(self, x):
+        """
+        Applies the forward pass of the convolutional layer.
+
+        Arguments
+        ---------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, channels, sequence_length).
+
+        Returns
+        -------
+        torch.Tensor
+            The output tensor after applying the convolution operation.
+            If `causal` is True, the input tensor is padded to ensure that
+            the output at each timestep only depends on the current and previous inputs.
+        """
+        if self.causal:
+            x = F.pad(x.unsqueeze(2), (self.left_padding, 0, 0, 0)).squeeze(2)
+
+        return super(Conv1d, self).forward(x)
+
+
+class ConvTranspose1d(nn.ConvTranspose1d):
```

Comment:
If I use the SB downsampling, it causes an error when loading the pretrained weights since the naming could be a bit different.  Since it is going to be moved to the integration folder, I suggest keep all their module as it is . 
I am working on another recepie that include many of these quantization techniques.. i will take care of this in that PR.

Reply:
same as prev one.